### ISSUE TITLE
Request textDocument/documentSymbol failed

### ISSUE THREAD

**Issue by cig0** [VSChina/vscode-ansible#71]
Hi,

I get the following error whenever I'm editing an Ansible role, inventory or playbook:

```
[Error - 21:40:53] Request textDocument/documentSymbol failed.
  Message: Request textDocument/documentSymbol failed with message: Cannot read property 'push' of undefined
  Code: -32603 

```

![screen shot 2018-02-06 at 21 41 46-compressed](https://user-images.githubusercontent.com/394089/35892162-d4a9b71c-0b86-11e8-8542-628f43ed4040.png)

Thanks

**Comment by yaweiw** [comment]
@yungez

**Comment by yungez** [comment]
@cig0  thanks for reporting the issue. we're looking at it and update the result here.

**Comment by yungez** [comment]
@cig0  after investigation, the error is reproduced. when yaml file is invalid format, such wrong indent etc, extension failed to parse yaml file then this error. we've fixed the code to show nice and clear error message to user. the fix will be shipped in next release, estimate around end of Feb. Stay tuned.

**Comment by oestoer** [comment]
I've got a similar error if the yaml indenting is not correct, just for hover
```
[Error - 16:03:40] Request textDocument/hover failed.
  Message: Request textDocument/hover failed with message: Cannot read property 'push' of undefined
  Code: -32603 
```
Fixing the formatting fixes the issue too. Thought to log it anyway.

**Comment by kkolk** [comment]
Could we just suppress this error completely?  I don't see any evidence on whatever it's alerting on is affecting the functionality of the runbook.  Having my terminal switches to Output anytime, I paste content into a runbook because it needs to output this message is frustrating.

**Comment by cig0** [comment]
Indeed it would be nice to have a switch to let the user completely turn off this functionality to avoid wasting CPU cycles and memory when not needed.

**Comment by yungez** [comment]
fixed by PR #69 . released in 0.1.4. pls have a try and let us know if any further question. thanks.

**Comment by superiums** [comment]
suffering this issue again.

reported by volar:

```
[Error - 16:40:29] Request textDocument/foldingRange failed.
  Message: Request textDocument/foldingRange failed with message: host.fileExists is not a function
  Code: -32603 
```
